### PR TITLE
Support for refresh without id_token, run silent renew using only the access token

### PR DESCRIPTION
**Pull request by FabianGosebrink**
- Support refresh tokens  without returning an id_token in the refresh
- run silent renew using only the access token
- id_token only has to be valid on the first authentication
- add support to disable id_token validation completely, not recommended

Fixing #1552 and #1408

### CODE REVIEW

File: docs/site/angular-auth-oidc-client/docs/documentation/02-configuration.md [hunk 1]
```diff
@@ -409,12 +409,12 @@ Extra parameters that can be added to the token URL request.
...
 Disables the `auth_time` validation for `id_tokens` in a refresh due to Microsoft Azure's incorrect implementation.
 
-### `enableIdTokenExpiredValidationInRenew`
+### `triggerRefreshWhenIdTokenExpired`
 
 - Type: `boolean`
 - Required: `false`
 
-Enables the `id_token` validation. You can disable this validation if you would like to ignore expired values during the renew process. If no `id_token` is returned while using refresh tokens, set this to false. <br/>
+Enables the `id_token` validation. You can disable this validation if you would like to ignore expired values during the renew process or after the first renew and in the expiry check. The renew will only be triggered if the access token expires. If no `id_token` is returned while using refresh tokens, set this to false. <br/>
```
Comment (FabianGosebrink) [inline docs/site/angular-auth-oidc-client/docs/documentation/02-configuration.md:417]: This text is not correct or confusing. This setting does not enable the id_token validation. It enables if a refresh should be triggered when the id token is expired IIRC. Please update text accordingly.

File: docs/site/angular-auth-oidc-client/docs/documentation/02-configuration.md [hunk 2]
```diff
@@ -467,3 +467,12 @@ Allows multiple usage of refresh token. Refresh tokens which can be stored safel
 Activate this property only if your OIDC provider cannot be configured to rotate refresh tokens.
 
 Default = _false_
+
+### `disableIdTokenValidation`
+
+- Type: `boolean`
+- Required: `false`
+
+Disable validation for id_token. This is not recommended! You should always validate the id_token if returned.
```
Comment (FabianGosebrink) [inline docs/site/angular-auth-oidc-client/docs/documentation/02-configuration.md:476]: This does NOT disable the id token validation completely. The id token is still checked once in the beginning and after a refresh of the page This has to be mentioned here. What happens when the id token is expired after a refresh? Mention here, too.

File: docs/site/angular-auth-oidc-client/docs/documentation/12-silent-renew.md [hunk 3]
```diff
@@ -18,6 +18,39 @@ For session renewal with the PKCE flow, no iframes are used by default. Instead,
...
+        redirectUrl: window.location.origin,
+        postLogoutRedirectUri: window.location.origin,
+        clientId: '--client_id--',
+        scope: 'openid profile offline_access',
+        responseType: 'code',
+        silentRenew: true,
+        useRefreshToken: true,
+        ignoreNonceAfterRefresh: true, // this is required if the id_token is not returned
+        triggerRefreshWhenIdTokenExpired: false, // required to refresh the browser if id_token is not updated after the first authentication
+        // allowUnsafeReuseRefreshToken: true, // this is required if the refresh token is not rotated
```
Comment (FabianGosebrink) [inline docs/site/angular-auth-oidc-client/docs/documentation/12-silent-renew.md:43]: Remove this commented out line

File: docs/site/angular-auth-oidc-client/docs/documentation/12-silent-renew.md [hunk 4]
```diff
@@ -18,6 +18,39 @@ For session renewal with the PKCE flow, no iframes are used by default. Instead,
...
+        authority: '--idp--',
+        redirectUrl: window.location.origin,
+        postLogoutRedirectUri: window.location.origin,
+        clientId: '--client_id--',
+        scope: 'openid profile offline_access',
+        responseType: 'code',
+        silentRenew: true,
+        useRefreshToken: true,
+        ignoreNonceAfterRefresh: true, // this is required if the id_token is not returned
+        triggerRefreshWhenIdTokenExpired: false, // required to refresh the browser if id_token is not updated after the first authentication
```
Comment (FabianGosebrink) [inline docs/site/angular-auth-oidc-client/docs/documentation/12-silent-renew.md:42]: This comment is confusing. What are you trying to say?

File: docs/site/angular-auth-oidc-client/docs/documentation/12-silent-renew.md (hunk 3 above)
Comment (damienbod) [reply docs/site/angular-auth-oidc-client/docs/documentation/12-silent-renew.md:43]: this is for the docs, some IDPs require this.

File: docs/site/angular-auth-oidc-client/docs/documentation/02-configuration.md (hunk 1 above)
Comment (damienbod) [reply docs/site/angular-auth-oidc-client/docs/documentation/02-configuration.md]: fixed

**Pull request merged**